src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,107 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold = 0;
+    private int initialCapacity = 0;
+
+    private class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+        initialCapacity = DEFAULT_INITIAL_CAPACITY;
+    }
+
     @Override
     public void put(K key, V value) {
+        int hash = getHash(key);
+        
+        if (size > threshold) {
+            resize();
+        }
+
+        if (table[hash] == null) {
+            table[hash] = new Node<>(hash, key, value, null);
+            size++;
+        } else {
+            Node<K, V> searchNode = table[hash];
+
+            while (searchNode != null) {
+                if (equals(searchNode.key, key)) {
+                    searchNode.value = value;
+                    searchNode = null;
```

Comment:
Why should we assign null to searchNode?

Reply:
to end loop. Is better to use break?
I have no idea how to refactor this while :(